pkg/skaffold/docker/parse.go

```diff
@@ -56,8 +65,12 @@ func expandBuildArgs(nodes []*parser.Node, buildArgs map[string]*string) {
 
 		// build arg's value
 		var value string
+		var err error
 		if buildArgs[key] != nil {
-			value = *buildArgs[key]
+			value, err = evaluateBuildArgsValue(*buildArgs[key])
+			if err != nil {
+				logrus.Errorf("unable to get value for build arg %q", key)
```

Comment:
We should fail the build if the template is invalid instead of just logging - please propagate back the err value through    `expandBuildArgs`,  `readDockerfile` up to `GetDependencies`.